universe/Conditions.cpp

```diff
@@ -10311,8 +10311,13 @@ And::And(std::unique_ptr<Condition>&& operand1, std::unique_ptr<Condition>&& ope
     Condition()
 {
     // would prefer to initialize the vector m_operands in the initializer list, but this is difficult with non-copyable unique_ptr parameters
-    if (operand1)
-        m_operands.push_back(std::move(operand1));
+    if (operand1) {
```

Comment:
@o01eg any reason not to also consider other subconditions besides `operand1` ?

Reply:
Other cases are rare as we don't use nested And or Or in FOCS. I could add it in other PRs. Maybe it means we don't need separate flattening function.